lib/appsignal/oban.ex

```diff
@@ -0,0 +1,172 @@
+        %{job: job},
+        _config
+      ) do
+    span = @tracer.create_span("oban")
+
+    span
+    |> @span.set_name(to_string(job.worker))
+    |> @span.set_sample_data("params", job.args)
+    |> @span.set_attribute("id", job.id)
+    |> @span.set_attribute("queue", to_string(job.queue))
+    |> @span.set_attribute("attempt", job.attempt)
+    |> @span.set_attribute("priority", job.priority)
+    |> @span.set_attribute("appsignal:category", "job.oban")
+
+    for {key, value} <- job.tags do
+      @span.set_attribute(span, "job_tag_#{key}", value)
+    end
+
+    increment_job_start_counter(job)
+
+    add_job_queue_time_value(job)
+  end
+
+  def oban_job_stop(
+        _event,
+        %{duration: duration},
+        %{state: state, result: result, job: job},
+        _config
+      ) do
+    @tracer.current_span()
+    |> @span.set_attribute("state", to_string(state))
+    |> @span.set_attribute("result", inspect(result))
+    |> @tracer.close_span()
+
+    increment_job_stop_counter(job, state)
+
+    add_job_duration_value(job, duration, state)
+  end
+
+  def oban_job_exception(_event, %{duration: duration}, metadata, _config) do
+    span =
+      @tracer.current_span()
+      |> @span.set_attribute("state", to_string(metadata[:state]))
+      |> @span.set_attribute("worker", to_string(metadata[:job].worker))
+
+    if Map.has_key?(metadata, :kind) and Map.has_key?(metadata, :reason) and
+         Map.has_key?(metadata, :stacktrace) do
+      @span.add_error(span, metadata[:kind], metadata[:reason], metadata[:stacktrace])
+    end
+
+    @tracer.close_span(span)
+
+    increment_job_stop_counter(metadata[:job], metadata[:state])
+
+    add_job_duration_value(metadata[:job], duration, metadata[:state])
+  end
+
+  def oban_insert_job_start(_event, _measurements, metadata, _config) do
+    span = @tracer.create_span("oban", @tracer.current_span)
+
+    @span.set_attribute(span, "appsignal:category", "insert_job.oban")
+
+    case metadata[:changeset] do
+      %{changes: %{worker: worker}} ->
+        @span.set_name(span, "Insert job (#{worker})")
+
+      _ ->
+        @span.set_name(span, "Insert job")
+    end
+  end
+
+  def oban_insert_job_stop(_event, _measurements, metadata, _config) do
+    case metadata[:changeset] do
+      %{changes: %{worker: worker}} ->
+        @appsignal.increment_counter("oban_job_insert", 1, %{worker: to_string(worker)})
+
+      _ ->
+        nil
+    end
+
+    @tracer.close_span(@tracer.current_span())
+  end
+
+  def oban_insert_job_exception(_event, _measurements, _metadata, _config) do
+    @tracer.close_span(@tracer.current_span())
+  end
+
+  defp increment_job_start_counter(job) do
+    tag_combinations = [
+      %{worker: to_string(job.worker), queue: to_string(job.queue)},
+      %{worker: to_string(job.worker)},
+      %{queue: to_string(job.queue)}
+    ]
+
+    Enum.each(tag_combinations, fn tags ->
+      @appsignal.increment_counter("oban_job_start", 1, tags)
```

Comment:
I'm a little confused what this metric tracks that `oban_job_start` doesn't, other than the missing queue tag. Is there a scenario when a job starts, but doesn't stop?

Reply:
I think you're right, `oban_job_start` is probably redundant. Removing it and adding the queue to the `oban_job_stop` tag combinations should do.

Reply:
In sidekiq the metric name is something like `sidekiq_queue_job_count`. That's a bit more generic than a metric called `_stop`. We track one count for every processed job `:status => :processed`, and a separate count with the tag `:status => :failed`.

Let's replicate that behavior, unless you see a problem with that?

Reply:
> should I just rename oban_job_stop to oban_job_count, and remove the oban_job_start metric?

Yes, I would say so.